### PR TITLE
T1913

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 2

Live output at: http://botio.butyr.org:8000/d96a4eb1499f289/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 3

Live output at: http://botio.butyr.org:8000/17476b633ffd3f6/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/d96a4eb1499f289/output.txt

Total script time: 0.56 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/17476b633ffd3f6/output.txt

Total script time: 0.78 mins

Server is running on http://pr752.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f3371237febe73b/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f3371237febe73b/output.txt

Total script time: 1.12 mins

File: templates/assets/plugins/repeat/popcorn.repeat.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        options.loop = options.loop || 0;
+        options.count = +options.loop;
+        options.toString = function() {
+          return "Loop: " + ( options.loop > 0 ? options.loop : "forever" );
+        };
+      },
+      start: function( event, options ) {
+      },
+      end: function( event, options ) {
+          console.log( _this, options.start, options.count, options.loop );
```
Comment (jbuck) [inline templates/assets/plugins/repeat/popcorn.repeat.js]: stay away assassino

File: templates/assets/plugins/repeat/popcorn.repeat.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    var _this = this;
+    return {
+      _setup: function( options ) {
+        options.loop = options.loop || 0;
+        options.count = +options.loop;
+        options.toString = function() {
+          return "Loop: " + ( options.loop > 0 ? options.loop : "forever" );
+        };
+      },
+      start: function( event, options ) {
```
Comment (jbuck) [inline templates/assets/plugins/repeat/popcorn.repeat.js]: Is it possible to remove this empty start() function?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/ebb46f83aabb4be/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/ebb46f83aabb4be/output.txt

Total script time: 0.70 mins

Server is running on http://pr752.butyr.org/templates/basic/
Run the Test Suite: http://pr752.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/789341cade7248d/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/789341cade7248d/output.txt

Total script time: 0.60 mins

Server is running on http://pr752.butyr.org/templates/basic/
Run the Test Suite: http://pr752.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/2be8581cc7c0170/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/970ce447e2cd6da/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/2be8581cc7c0170/output.txt

Total script time: 0.97 mins

Server is running on http://pr752.butyr.org/templates/basic/
Run the Test Suite: http://pr752.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/970ce447e2cd6da/output.txt

Total script time: 1.01 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/ccebee04e50b883/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/ccebee04e50b883/output.txt

Total script time: 0.86 mins

Server is running on http://pr752.butyr.org/templates/basic/
Run the Test Suite: http://pr752.butyr.org/test/

**Pull request merged**